dhcpv4/bsdp/option_vendor_specific_information.go

```diff
@@ -11,20 +9,17 @@ import (
 // OptVendorSpecificInformation encapsulates the BSDP-specific options used for
 // the protocol.
 type OptVendorSpecificInformation struct {
-	Options []dhcpv4.Option
+	Options dhcpv4.Options
```

Comment:
Not sure that I like this particular type alias - what's wrong with just having it as a slice? it's more clear that it's a slice of a struct/interface, rather than having to look at what a `dhcpv4.Options` is.

Reply:
So here's the deal. You've got the []dhcpv4.Option type in a couple of places: DHCPv4, OptRelayAgentInfo, and here.

A dhcpv4.Options type can be the one place to implement the following:
```go
func (Options) Get(...)
func (Options) Has(...)
func (Options) Add(...)
func (Options) Update(...)
```

rather than reimplementing GetOption and GetOneOption on multiple types and having the OptionGetter interface, just implement it on Options and spread Options around.

Reply:
I.e. there is no need to have both https://github.com/insomniacslk/dhcp/blob/108ed92e1c9901936541020bc3214533acce77bb/dhcpv4/bsdp/option_vendor_specific_information.go#L135 and https://github.com/insomniacslk/dhcp/blob/108ed92e1c9901936541020bc3214533acce77bb/dhcpv4/dhcpv4.go#L366

and eventually you'll start replicating HasOption and UpdateOption and AddOption to OptVendorSpecificInformation as well as the relay agent stuff... so an Options type saves you that work.

Reply:
yeah I get how this works, and in an isolated context it makes perfectly sense from a compactness perspective. I think it introduces a bit more confusion, but it's probably acceptable compared to the benefits of a better / more compact interface. As long as we use always `Options` instead of `[]Option` I guess this is OK. If we don't like it at any stage it's always possible to go back. Let's go with `Options` now